### PR TITLE
Yield unpacked lists

### LINKED ISSUE #201: Classification result using pre-trained model doesn't really make a lot of sense
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

_No response_

### Bug

```
!yolo task=classify mode=predict model=yolov8n-cls.pt conf=0.25 source='https://media.roboflow.com/notebooks/examples/dog.jpeg'
```

download (11)

1. Looks like the model is operating on class ids, not human-readable class names.

2. Also saving inference results on images is cool, but is a bit useless. It is virtually impossible to work with that inference result in an automated way. It would be cool to save the result per image in a txt file for example. 

3. I'm also not sure how to interpret those values before id. Would it be possible to push them through some softmax so that they could be interpretable as probability? 

### Environment

Ultralytics YOLOv8.0.2 🚀 Python-3.8.16 torch-1.13.0+cu116 CUDA:0 (Tesla T4, 15110MiB)
Setup complete ✅ (2 CPUs, 12.7 GB RAM, 23.1/78.2 GB disk)

Google Colab

### Minimal Reproducible Example

```
!yolo task=classify mode=predict model=yolov8n-cls.pt conf=0.25 source='https://media.roboflow.com/notebooks/examples/dog.jpeg'
```

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by AyushExel**
Fixes https://github.com/ultralytics/ultralytics/issues/201#issuecomment-1398476302
@Laughing-q @glenn-jocher can you take look to confirm..
yield from basically in this case is like doing `yield result for result in results` 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improved streaming inference support in the prediction engine.

### 📊 Key Changes
- Added a new test case to verify model predictions in streaming mode.
- Updated the stream inference method to yield individual results instead of a list of results.
- Simplified the merging of stream output in the `__call__` method of the prediction engine.

### 🎯 Purpose & Impact
- Ensures that streaming predictions function correctly, improving real-time performance. 💨
- Allows users to process video streams or sequences of images more efficiently. 📹
- The test update helps prevent future regressions in streaming functionality. 🛡️
- This change benefits applications that require immediate processing of each frame or prediction, such as real-time monitoring systems. 🚀

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@AyushExel should this be merged into 8.0.12 deploying soon?
